components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/impl/FederatedAuthenticatorConfigBuilderFactory.java

```diff
@@ -141,6 +140,38 @@ public static FederatedAuthenticator build(FederatedAuthenticatorConfig config)
 
         return federatedAuthenticator;
     }
+
+    /**
+     * Builds a list of FederatedAuthenticatorListItem instances based on the given array of
+     * FederatedAuthenticatorConfig.
+     *
+     * @param fedAuthConfigs Array of FederatedAuthenticatorConfig instances.
+     * @param idpResourceId  Identity provider resource ID.
+     * @return List of FederatedAuthenticatorListItem instances.
+     */
+    public static List<FederatedAuthenticatorListItem> build(FederatedAuthenticatorConfig[] fedAuthConfigs,
+                                                             String idpResourceId) {
+
+        List<FederatedAuthenticatorListItem> authenticators = new ArrayList<>();
+        for (FederatedAuthenticatorConfig config : fedAuthConfigs) {
+            FederatedAuthenticatorListItem authenticatorListItem = new FederatedAuthenticatorListItem();
+            authenticatorListItem.setAuthenticatorId(base64URLEncode(config.getName()));
+            authenticatorListItem.setName(config.getName());
+            authenticatorListItem.setIsEnabled(config.isEnabled());
+            authenticatorListItem.definedBy(FederatedAuthenticatorListItem.DefinedByEnum.valueOf(
+                    config.getDefinedByType().toString()));
+            List<String> tags = resolveAuthenticatorTags(config);
+            if (tags.isEmpty()) {
```

Comment:
As we are only setting the tags when it is not empty, in other cases tags in authenticatorListItem remain null. 
By considering this, we can set the resolveAuthenticatorTags() method to return a list when it is not empty and null on other cases.